gen/templates/generic/data_source.go

```diff
@@ -60,7 +60,11 @@ func (d *{{camelCase .Name}}DataSource) Schema(ctx context.Context, req datasour
 		Attributes: map[string]schema.Attribute{
 			"id": schema.StringAttribute{
 				MarkdownDescription: "The id of the object",
-				Required:            true,
+				{{- if not .RemoveId}}
```

Comment:
Has nothing to do with this line, but we should also update the "data_source.tf" template to render a correct example for data sources with "remove_id", ie not including "id" as an attribute, but rather the attributes marked as "query_param".

Reply:
Yea that is probably a good idea